ios/Classes/PspdfkitPlugin.m

```diff
@@ -307,6 +314,56 @@ - (PSPDFPageIndex)pageIndex:(NSDictionary *)dictionary {
     return (PSPDFPageIndex)[dictionary[@"startPage"] unsignedLongValue];
 }
 
+- (PSPDFPageMode)pageMode:(NSDictionary *)dictionary {
+    PSPDFPageMode pageMode = PSPDFConfiguration.defaultConfiguration.pageMode;
+
+    if ((id)dictionary == NSNull.null || !dictionary || dictionary.count == 0) {
+        return pageMode;
+    }
+
+    NSString *value = dictionary[@"pageLayoutMode"];
+    if (value) {
+        if ([value isEqualToString:@"automatic"]) {
+            pageMode = PSPDFPageModeAutomatic;
+        } else if ([value isEqualToString:@"single"]) {
+            pageMode = PSPDFPageModeSingle;
+        } else if ([value isEqualToString:@"double"]) {
+            pageMode = PSPDFPageModeDouble;
+        }
+    }
+    return pageMode;
+}
+
+- (PSPDFSettingsOptions)settingsOptions:(nullable NSArray <NSString *> *)options {
+    if ((id)options == NSNull.null || !options || options.count == 0) {
+        return PSPDFSettingsOptionDefault;
+    }
+
+    PSPDFSettingsOptions finalOptions = 0;
+
+    for (NSString *option in options) {
+        if ([option isEqualToString:@"scrollDirection"]) {
+            finalOptions |= PSPDFSettingsOptionScrollDirection;
+        } else if ([option isEqualToString:@"pageTransition"]) {
+            finalOptions |= PSPDFSettingsOptionPageTransition;
+        } else if ([option isEqualToString:@"appearance"]) {
+            finalOptions |= PSPDFSettingsOptionAppearance;
+        } else if ([option isEqualToString:@"brightness"]) {
+            finalOptions |= PSPDFSettingsOptionBrightness;
+        } else if ([option isEqualToString:@"pageMode"]) {
+            finalOptions |= PSPDFSettingsOptionPageMode;
+        } else if ([option isEqualToString:@"spreadFitting"]) {
+            finalOptions |= PSPDFSettingsOptionSpreadFitting;
+        }
```

Comment:
I wonder if we should log an error or even assert if invalid values are present in the options array.

Reply:
Speaking of the options array, any reason why we're using an array instead of a dictionary?

Reply:
> I wonder if we should log an error or even assert if invalid values are present in the options array.

I think that invalid values should be ignored without asserting. We will simply log that an invalid value was passed and we will ignore it.

> Speaking of the options array, any reason why we're using an array instead of a dictionary?

After taking a closer look into the implementation, it makes sense to use an array because the Dart payload will be nicer.